### PR TITLE
[aws-lambda] add TResult as generic parameter and pass it to Handler<Args, Result>

### PR DESCRIPTION
**Pull request by zirkelc**
This PR adds `TResult` as generic parameter for `AmplifyGraphQlResolverHandler<TArguments, TSource, TResult>` and pass it to the `Handler<TEvent, TResult>`.

This allows to specify the return type of a GraphQL resolver handler and is consistent to the existing `AppSyncResolverHandler<TArguments, TResult, TSource>`, that also defines `TResult` and passes it to `Handler`. 

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a9f480c82b3b356990e5dd574bf2c1ba9c2fdfb5/types/aws-lambda/handler.d.ts#L84-L88

### CODE REVIEW

File: types/aws-lambda/trigger/appsync-resolver.d.ts
Comment (simonbuchan) [inline types/aws-lambda/trigger/appsync-resolver.d.ts:68]: What are these? `info` already seemingly has these and they're not covered by the tests.

**Review by simonbuchan — COMMENTED** [review]
Other than the changes to `AppSyncResolverEvent` this seems pretty innocuous.

**Comment by simonbuchan** [comment]
Not sure what's up when CI - looks like an expected error in a dependant package, but I can't see how that would be.

File: types/aws-lambda/trigger/appsync-resolver.d.ts
Comment (zirkelc) [reply types/aws-lambda/trigger/appsync-resolver.d.ts]: Sorry, I accidentally left these changes in the commit. These two fields are already available in `AmplifyGraphQlResolverEvent` and therefore, must not be added to `AppSyncResolverEvent`.

I reverted changes to this file.

**Comment by zirkelc** [comment]
@simonbuchan I can't figure out the reason the CI fails. I can see that the package `lambda-tester` is causing the CI to fail. On my local machine, running `npm test lambda-tester` doesn't bring up any errors.

**Review by simonbuchan — APPROVED** [review]
Assuming CI is just some weird flake or something, looks good to go from me!

**Review by jamesorlakin — APPROVED** [review]
Good spot - I'm not sure why I didn't think to do that when making the Amplify handler!

**Comment by zirkelc** [comment]
@jamesorlakin @simonbuchan sorry to bother you, I'd like to know when the new package will be released?

**Comment by simonbuchan** [comment]
Merges and publishing are automated, but they're blocked by CI failing. I was hoping it was a transient issue due to infrastructure changes that would get fixed (it's happened a few times before), but I might need to poke at this tomorrow, see if I can fix the broken test or if I need to poke one of the DT maintainers.

**Comment by zirkelc** [comment]
@simonbuchan I created a PR for package `lambda-tester` that should hopefully unblock this PR:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64048

**Comment by zirkelc** [comment]
The tests keep failing even though I changed the tests on `lambda-tester`. Event if I remove all tests inside `types/lambda-tester/lambda-tester-tests.ts`, the tests fail:

```
Testing 1 dependent packages: lambda-tester
Installing NPM dependencies...
1> aws-lambda START
2> lambda-tester START
3> lambda-tester START
dtslint@0.0.142
dtslint@0.0.142
dtslint@0.0.142
2> lambda-tester passed, but was expected to fail.
3> lambda-tester passed, but was expected to fail.
```

Could it be that the failing test is actually related to the original NPM package and not the test case inside this repo?

**Comment by simonbuchan** [comment]
Thanks for looking into this after I dropped the ball :|.

I've cloned and `npm run test-all` runs cleanly on master (though I think only because it diffs against master for code to re-run) and though it has several test errors on this PR it all looks like TS 5.0 errors. It's unclear why the output is so different from CI...

@sandersn as the last updater on the infrastructure pinned issue #44317, any idea how to approach fixing this?

**Comment by sandersn** [comment]
lambda-tester was known to fail (for some reason, I haven't looked at the PR at that added it). But it's fixed now and no longer on the list of expected failures. Re-running CI should succeed now.

**Comment by simonbuchan** [comment]
@sandersn Thanks, good to know re-opening is a recommendation for kicking CI. Since you're here already, would you mind kicking #63844 too?

**Pull request merged**
